DataFormats/HGCDigi/interface/PreMixHGCSimAccumulator.h

```diff
@@ -0,0 +1,162 @@
+#ifndef DataFormats_HGCDigi_PreMixHGCSimAccumulator_h
+#define DataFormats_HGCDigi_PreMixHGCSimAccumulator_h
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include <iostream>
+#include <vector>
+#include <cassert>
+
+class PreMixHGCSimAccumulator {
```

Comment:
Yes, the PreMixHGCSimAccumulator class is intended to replace PHGCSimAccumulator. But I chose to take a conservative step and keep it lie around. I would feel it safer to do away with it in my next project(which itself involves potential improvement of the PreMixHGCSimAccumulator class) totally.

Reply:
I'm not sure I understand. With this PR `PHGCSimAccumulator` becomes completely unused. The implementation can still be looked for in the git history, so I don't see what value keeping it would give (but keeping it can lead to confusion). 

> As this is just development version and not a production version,

This PR *does* make `PreMixHGCSimAccumulator` as the production version for HGCal premixing (or as close to production as the Phase2 premixing in general is).

Reply:
> @adas1994 Have you checked how much the PU200 pileup library compressed disk size increases with this PR? (basically `edmEventSize` of the pileup library made for e.g. 100 events before and after the PR)

No. I haven't checked that. And I will remove the PHGCSimAccumulator class.